annotations/update_pending_annotations.py

```diff
@@ -0,0 +1,152 @@
+import synapseclient
+from synapseclient import Table, RowSet
+import argparse
+import pandas as pd
+from attribute_dictionary import PUBLICATION_DICT
+
+
+### Login to Synapse ###
+def login():
+
+    syn = synapseclient.Synapse()
+    syn.login()
+
+    return syn
+
+
+def get_args():
+
+    parser = argparse.ArgumentParser(
+        description=
+        'Get synapse table id of annotations to be editd and synapse table id of controlled vocabulary mappings'
+    )
+    parser.add_argument(
+        '-t',
+        '--table_id',
+        type=str,
+        default='syn21868591',
+        help=
+        'Synapse table id where annotations will be updated. Probably the Publications Merged table.'
+    )
+    parser.add_argument('-m',
+                        '--manifest_path',
+                        type=str,
+                        help='Path where updated manifest is stored.')
+    return parser.parse_args()
+
+
+def get_updated_df(manifest):
+
+    updated_df = pd.read_csv(manifest, index_col=False).fillna("")
+
+    return updated_df
+
+
+def get_annotations(table_id, updated_df, syn):
+
+    pubmed_list = updated_df['Pubmed Id'].astype(str).tolist()
+    pubmed_string = ', '.join(pubmed_list)
+    print(f'list of pumbmeds to be updated: {pubmed_string}')
+
+    annots_query = syn.tableQuery(
+        f"SELECT pubMedId, assay, tumorType, tissue, dataset, accessibility FROM {table_id} WHERE pubMedId IN ({pubmed_string})"
+    )
+
+    return annots_query
+
+
+def edit_annotations(updated_df, annots_query, syn, table_id):
```

Comment:
There's a lot going on in this function, so a docstring here to describe its general purpose would be useful!